playwright/src/main/java/com/microsoft/playwright/impl/Utils.java

```diff
@@ -412,4 +414,40 @@ static int fromJsRegexFlags(String regexFlags) {
   static String addSourceUrlToScript(String source, Path path) {
     return source + "\n//# sourceURL=" + path.toString().replace("\n", "");
   }
+
+  static void addToProtocol(JsonObject params, List<ClientCertificate> clientCertificateList) {
+    if (clientCertificateList == null) {
+      return;
+    }
+    JsonArray clientCertificates = new JsonArray();
+    for (ClientCertificate cert: clientCertificateList) {
+      JsonObject jsonCert = new JsonObject();
+      jsonCert.addProperty("origin", cert.origin);
+      try {
+        if (cert.certPath != null) {
+          byte[] bytes = readAllBytes(cert.certPath);
+          String base64 = Base64.getEncoder().encodeToString(bytes);
+          jsonCert.addProperty("cert",  base64);
+        }
+        if (cert.keyPath != null) {
+          byte[] bytes = readAllBytes(cert.keyPath);
+          String base64 = Base64.getEncoder().encodeToString(bytes);
+          jsonCert.addProperty("key", base64);
+        }
+        if (cert.pfxPath != null) {
+          byte[] bytes = readAllBytes(cert.pfxPath);
+          String base64 = Base64.getEncoder().encodeToString(bytes);
+          params.addProperty("pfx", base64);
+        }
+      } catch (IOException e) {
+        throw new PlaywrightException("Failed to read from file", e);
+      }
+      if (cert.passphrase != null) {
+        jsonCert.addProperty("passphrase", cert.passphrase);
+      }
+      clientCertificates.add(jsonCert);
+    }
+    params.remove("clientCertificates");
```

Comment:
Not that I mind, but it's surprising there could already be a property with this name.

Reply:
That's because we use `new Gson().toJSON(options)`, we can either nullify the option before the call or remove it from JSON like here.